### ISSUE TITLE
Applying ns3-ai to multiple nodes

### ISSUE THREAD

**Issue by Necropsy** [hust-diangroup/ns3-ai#32]
Hello, initially congratulations for the project developed!

I am a master's student and I am developing a project that aims to improve the AODV protocol using machine learning. I would like to know if it is possible to use the ns3-ai module in the AODV protocol, where when defining it as a routing protocol for the network, each node will have an individual instance of ns3-ai. I would like each node in the network to be able to manage their own variables to control their AODV instance using their own neural network, is this possible with ns3-ai? Would you have any example of how this can be done (multiple instances of ns3-ai in parallel)?

**Comment by Mauriyin** [comment]
I think you can do that with ns3-ai. We will present a similar example for the WiFi-rate control algorithm in [wns3](https://www.nsnam.org/research/wns3/wns3-2021/program/) this year, and you are welcome to join us!

**Comment by Necropsy** [comment]
Hi @Mauriyin,

Thank you very much for the answer, I will participate yes, registration already done!

But in relation to this specific application, if I use ns3-ai directly in the AODV protocol, when defining the protocol in the simulation environment, each node would have an instance of the artificial intelligence module, however when executing the code in python containing a neural network I would only be executing a specific network and not each individual network of nodes, correct? (note: in this project I would have to implement one for each node and running directly within the routing protocol, a central network collecting information and identifying the network nodes by id would not be an option)

**Comment by Mauriyin** [comment]
I see. I think you can establish different processes with python, each one corresponding to one router, and use different memory id to transfer data to the neural net. So basically you need to construct a map like this:
in ns-3
```
node 0 data -> mem 0 -> python 0
node 1 data -> mem 1 -> python 1
node 2 data -> mem 2 -> python 2
....
```
and then you can have some NN running in different python programs.

**Comment by Guy122** [comment]
> I see. I think you can establish different processes with python, each one corresponding to one router, and use different memory id to transfer data to the neural net. So basically you need to construct a map like this: in ns-3
> 
> ```
> node 0 data -> mem 0 -> python 0
> node 1 data -> mem 1 -> python 1
> node 2 data -> mem 2 -> python 2
> ....
> ```
> 
> and then you can have some NN running in different python programs.
 
Tried doing that and it works, but what I want to do is create a single Python code that populates all the memories like this:

node 0 data -> mem 0 -> python 0
node 1 data -> mem 1 -> python 0
node 2 data -> mem 2 -> python 0
....

Is it feasible to implement?
